homeassistant/scripts/influxdb_migrator.py

```diff
@@ -0,0 +1,177 @@
+        '--script',
+        choices=['influxdb_migrator'])
+
+    args = parser.parse_args()
+
+    # Get client for old DB
+    client = InfluxDBClient(args.host, args.port,
+                            args.username, args.password)
+    client.switch_database(args.dbname)
+    # Get DB list
+    db_list = [db['name'] for db in client.get_list_database()]
+    # Get measurements of the old DB
+    res = client.query('SHOW MEASUREMENTS')
+    measurements = [measurement['name'] for measurement in res.get_points()]
+    nb_measurements = len(measurements)
+    # Move data
+    # Get old DB name
+    old_dbname = "{}__old".format(args.dbname)
+    # Create old DB if needed
+    if old_dbname not in db_list:
+        client.create_database(old_dbname)
+    # Copy data to the old DB
+    client.query('''SELECT * INTO {}..:MEASUREMENT FROM /.*/ '''.format(old_dbname))
+
+    # Delete the database
+    client.drop_database(args.dbname)
+    # Create new DB if needed
+    client.create_database(args.dbname)
+    client.switch_database(old_dbname)
+    # Get client for new DB
+    new_client = InfluxDBClient(args.host, args.port, args.username,
+                                args.password, args.dbname)
+    # Counter of points without time
+    point_wt_time = 0
+
+    print("Migrating from {} to {}".format(old_dbname, args.dbname))
+    # Walk into measurenebt
+    for index, measurement in enumerate(measurements):
+        # Print progess
+        print_progress(index, nb_measurements)
+        # Get tag list
+        res = client.query('''SHOW TAG KEYS FROM "{}"'''.format(measurement))
+        tags = [v['tagKey'] for v in res.get_points()]
+        # Get field list
+        res = client.query('''SHOW FIELD KEYS FROM "{}"'''.format(measurement))
+        fields = [v['fieldKey'] for v in res.get_points()]
+        # Get points, convert and send points to the new DB
+        offset = 0
+        while True:
+            nb_points = 0
+            # Prepare new points
+            new_points = []
+            # Get points
+            res = client.query('SELECT * FROM "{}" LIMIT {} OFFSET '
+                               '{}'.format(measurement, args.step, offset))
+            for point in res.get_points():
+                new_point = {"measurement": measurement,
+                             "tags": {},
+                             "fields": {},
+                             "time": None}
+                # Check time
+                if point["time"] is None:
+                    # Point without time
+                    point_wt_time += 1
+                    print("Can not convert point without time")
+                    continue
+                # Convert all fields
+                for field in fields:
+                    if isinstance(point[field], (int, bool)):
+                        new_point["fields"][field] = float(point[field])
+                    else:
+                        new_point["fields"][field] = point[field]
```

Comment:
You should replicate the behavior of the new code in the influxdb component. Values that can't be converted to float should be stored under field "state" for states and under "[attribute]_str" for state attributes. So I think you need have a special case for field == "value" to identify state vs state attributes.